### ISSUE TITLE
azurerm_storage_account.ip_rules doesn't take valid ip addresses

### ISSUE THREAD

**Issue by ryanhagen** [hashicorp/terraform-provider-azurerm#21169]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.2

### AzureRM Provider Version

3.42.0

### Affected Resource(s)/Data Source(s)

azurerm_storage_account

### Terraform Configuration Files

```hcl
resource "azurerm_storage_account" "docs" {
  name                = "docs"
  resource_group_name = data.azurerm_resource_group.rg.name

  location                 = data.azurerm_resource_group.rg.location
  account_tier             = "Standard"
  account_replication_type = "LRS"

  network_rules {
    default_action             = "Deny"
    ip_rules                   = split(",", file("./github_ips"))
    virtual_network_subnet_ids = [data.azurerm_subnet.subnet.id]
  }

  tags = local.common_tags
}
```

### Debug Output/Panic Output

```shell
https://gist.githubusercontent.com/ryanhagen/b6e3a6932648b1065d46ebf84f8475f8/raw/438693c20c0d187b81a02964680683368aca870c/tfdebugstorageaccount
```

### Expected Behaviour

I expect it to not fail on any of the addresses listed in my CSV doc  [here](https://pastebin.com/Zz7RCAji). I have sanitized the CSV to not include any /31 or /32. I do not have any private ranges. I do not have any phantom whitespaces.

### Actual Behaviour

╷
│ Error: "network_rules.0.ip_rules.1487" must start with IPV4 address and/or slash, number of bits (0-30) as prefix. Example: 23.45.1.0/30.
│ 
│   with azurerm_storage_account.zephyrdocs,
│   on zephyrdocs.tf line 11, in resource "azurerm_storage_account" "zephyrdocs":
│   11:     ip_rules                   = split(",", file("./github_ips"))

### Steps to Reproduce

terraform plan

### Important Factoids

Multi-tenant account shared across teams at my private org non-government

**Comment by tombuildsstuff** [comment]
@ryanhagen checking through the file, it looks like there's some private IP Ranges included in the list:

```
172.168.0.0/15
172.170.0.0/16
172.171.0.0/19
172.171.32.0/19
172.171.64.0/19
172.171.96.0/19
172.171.128.0/17
172.172.0.0/15
172.174.0.0/16
172.175.0.0/16
172.176.0.0/15
172.178.0.0/17
172.178.128.0/17
172.179.0.0/16
172.180.0.0/15
172.182.0.0/16
172.184.0.0/15
```

Which would be being caught by the validation here, since these (appear to be) private IP ranges - however I think the root cause of the confusion here is the lack of information in the validation message, which I'll PR a fix for shortly.

I recall reading that some reserved Private IP Ranges (which I thought were in the `192.` range) was made available for Public usage - so I'm wondering if the ranges listed above fall into this category?

**Comment by ryanhagen** [comment]
@tombuildsstuff 
Ive verified the 172 ranges above are not private based on IETF. The private 172 ranges are 172.16.0.0–172.31.255.255

**Comment by tombuildsstuff** [comment]
@ryanhagen so #21178 will call out which of the IP Ranges is causing this for you (which'll go out as a part of today's release) - [this logic](https://github.com/hashicorp/terraform-provider-azurerm/blob/8c617918fbba5755be72e3ae077dc9d6691ba180/internal/services/storage/validate/storage_account_ip_rule.go#L21) should handle the `172` range there, so I believe this should be clearer given changes coming in #21178.

**Comment by catriona-m** [comment]
It looks like this now been resolved by #21178 and released in v3.50.0 of the provider, so this can be closed now. Thanks!
